src/composables/transactionList.ts

```diff
@@ -0,0 +1,272 @@
+      transactions.value[address].pending[nodeNetworkId.value!] = [];
+    }
+    transactions.value[address].pending[nodeNetworkId.value!].push(transaction);
+  }
+
+  async function fetchTransactiosFromMiddleware(
+    transactionState: ITransactionsState,
+    address: Encoded.AccountAddress,
+    recent: boolean,
+    limit: number,
+  ) {
+    const { fetchFromMiddlewareCamelCased } = useMiddleware({ store });
+
+    const url = (recent || transactionState.nextPageUrl === '')
+      ? `/v2/accounts/${address}/activities?limit=${limit}`
+      : transactionState.nextPageUrl!;
+
+    try {
+      const { data, next } = await fetchFromMiddlewareCamelCased(url);
+      if (!recent) {
+        setTransactionsNextPage(address, next);
+      }
+      return data;
+    } catch (error) {
+      return [];
+    }
+  }
+
+  async function fetchPendingTransaction(
+    address: Encoded.AccountAddress,
+  ) {
+    const sdk = await getAeSdk();
+
+    try {
+      const fetchedPendingTransaction = (
+        await sdk.api.getPendingAccountTransactionsByPubkey(address)
+      );
+      const transactionState = getOrInitTransactionStateForAccount(address);
+      return JsonBig.parse(JsonBig.stringify(fetchedPendingTransaction))
+        .filter((transaction: ITransaction) => (
+          (transactionState.pending[nodeNetworkId.value!] || [])
+            .some(({ hash }) => transaction.hash === hash)
+        ));
+    } catch (error) {
+      return [];
+    }
+  }
+
+  async function fetchTipWithdrawnTransactions(address: Encoded.AccountAddress, recent: boolean) {
+    try {
+      const response = await fetchJson(
+        `${nodeNetworkId.value}/cache/events/?address=${address}&event=TipWithdrawn${recent ? '&limit=5' : ''}`,
+      );
+      if (response.message) return [];
+      const tipWithdrawnTransactions = (response as any[]).map(({
+        amount,
+        contract,
+        height,
+        data: { tx },
+        ...t
+      }) => ({
+        tx: {
+          ...tx,
+          address,
+          amount,
+          contractId: contract,
+          type: Tag[Tag.ContractCallTx],
+        },
+        ...t,
+        microTime: new Date(t.createdAt).getTime(),
+        blockHeight: height,
+        claim: true,
+      }));
+
+      return tipWithdrawnTransactions as ITransaction[];
+    } catch (eror) {
+      return [];
+    }
+  }
+
+  function removePendingTransactionByAccount(address: Encoded.AccountAddress, hash: string) {
+    const pendingTransactionForAccount: ITransaction[] = (
+      transactions.value[address]?.pending?.[nodeNetworkId.value!] || []
+    );
+
+    if (pendingTransactionForAccount.length) {
+      transactions.value[address].pending[nodeNetworkId.value!] = (
+        pendingTransactionForAccount.filter((transaction) => transaction.hash !== hash)
+      );
+    }
+  }
+
+  async function fetchTransactions(param: IFetchTransactionsParams): Promise<ITransaction[]> {
```

Comment:
I don't think an interface `IFetchTransactionsParams` is needed here. It could be just plain function arguments.